### DISCUSSION TITLE
Control multiple systems - one Moosh to rule them all

### DISCUSSION THREAD

**Discussion by halmaclean** [tmuras/moosh#449] [Ideas]
My company runs a moodle based system which is multi-tenanted - one set of Moodle code, multiple databases (one for each 'client'), and multiple 'moodledata' directories.

Is there a way to install Moosh and identify the 'client' we need to work on so that it runs only on that system? For example, client_7 is the database name, and we want to run some clean up on that client... all clients in the same system have access to the same moodle code, including plugins and blocks, etc, but each has a moodledata location that matches the database numbering - i.e. 'moodledata_7'. We want to be able to run a Moosh command to specify the relevant client 'id' and have it work on that setup only. Could this be as simple as editing the Moosh config to be able to identify the database and the path to the moodledata directory? We could, for example, create a way to 'setup' the Moosh session - run Moosh, set the ID for the system to work on, and for that session Moosh runs on that system. We would then need to be able to change the ID by running a command, and have Moosh run on that other system.

We also run multiple systems on the same server, some in the multi-tenanted system, and some as 'standalone' Moodle installs. For the standalone ones we imagine we add Moosh multiple times, but it would be far better to be able to identify which system we need to work on, and install only one set of Moosh code.

Any ideas for that?

Thanks in advance :)

**Comment by tmuras** [comment]
Standalone Moodle systems are simple. You install only one moosh and then either cd into Moodle www dir and run it from there, or you use moosh option that points to Moodle installation you want to manage:
	-p, --moodle-path=
		Moodle directory.

The simplest way to deal with your multi-tenancy would be to create temporary Moodle config.php file that points to the tenant you want to manage, then run moosh that (with -p option) points to directory with this config.php.

Other than that, moosh would need to be extended with some extra options, like (for example) forcing settings like DB credentials, data and www directories.

**Reply by davidpesce** [reply]
Why not use ansible?
